### PR TITLE
Remove dependency on dockerfile.build

### PR DESCRIPTION
**Pull request by PratikMahajan**
we are doing a multi-stage build in dockerfile.deploy to
remove the dependency on dockerfile.build

this is because we are installing certain modules in dockerfile.build
that are introducing vulnerabilities. We are not carrying these
vulnerabilities in the final container that we are shipping, we are just
using dockerfile.build container as a builder image. This can be achieved
in a multi-stage build, thus not requiring a separate build dockerfile.

### CODE REVIEW

File: dist/Dockerfile.e2e/Dockerfile
Comment (vrutkovs) [inline dist/Dockerfile.e2e/Dockerfile]: `/usr/bin/cincinnati-e2e-test: error while loading shared libraries: libssl.so.1.1: cannot open shared object file: No such file or directory`

I think we want `openssl`, not `-devel` here

Comment (PratikMahajan) [reply dist/Dockerfile.e2e/Dockerfile]: updated

Comment (vrutkovs) [reply dist/Dockerfile.e2e/Dockerfile]: Wrong container, these are installed in `builder`, not in resulting ubi8

**Comment by LalatenduMohanty** [comment]
@PratikMahajan Are you planing to remove the rpms coming from epel from https://github.com/openshift/cincinnati/blob/master/dist/Dockerfile.build/Dockerfile  after this PR merges?

**Review by vrutkovs — APPROVED** [review]
/lgtm

**Comment by vrutkovs** [comment]
/hold

Not sure why is it failing to start UWM. Lets see if a retest after 4 hrs helps

**Comment by vrutkovs** [comment]
/lgtm cancel

Now its a valid failure - seems `customrust` deploy image doesn't have `git2` installed, so git version is incorrectly being reported

File: dist/Dockerfile.e2e/Dockerfile
Comment (vrutkovs) [inline dist/Dockerfile.e2e/Dockerfile]: Previous `RUN` needs `libgit2` to fix e2e and report valid commit message

File: dist/Dockerfile.deploy/Dockerfile
Comment (vrutkovs) [inline dist/Dockerfile.deploy/Dockerfile]:
```suggestion
    yum -y install gcc openssl-devel protobuf-compiler jq skopeo buildah libgit2 && \
```

File: dist/Dockerfile.e2e/Dockerfile
Comment (vrutkovs) [reply dist/Dockerfile.e2e/Dockerfile]: I was wrong - `libgit2` must be present in build container, not necessary in e2e/deploy images

File: dist/Dockerfile.e2e/Dockerfile
Comment (vrutkovs) [inline dist/Dockerfile.e2e/Dockerfile:49]:
```suggestion
    yum install -y skopeo buildah && \
```

**Comment by vrutkovs** [comment]
That looks great! Lets squash the commits and I'll lgtm it

/hold cancel

**Review by vrutkovs — APPROVED** [review]
/lgtm

**Pull request merged**
